atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/pue/TimedConsensusForgettingStoreTest.java

```diff
@@ -0,0 +1,151 @@
+    private static final byte[] VALUE = PtBytes.toBytes("valyew");
+
+    private final ConsensusForgettingStore delegate = mock(ConsensusForgettingStore.class);
+    private final TaggedMetricRegistry registry = new DefaultTaggedMetricRegistry();
+    private final ConsensusForgettingStoreMetrics metrics = ConsensusForgettingStoreMetrics.of(registry);
+    private final AtomicInteger concurrentOperationTracker = new AtomicInteger(0);
+    private final ConsensusForgettingStore consensusForgettingStore =
+            new InstrumentedConsensusForgettingStore(delegate, metrics, concurrentOperationTracker);
+
+    @Before
+    public void before() {
+        metrics.concurrentCheckAndTouches(concurrentOperationTracker::get);
+    }
+
+    @Test
+    public void forwardsSimpleCallsToDelegate() throws ExecutionException, InterruptedException {
+        consensusForgettingStore.putUnlessExists(CELL, VALUE);
+        verify(delegate).putUnlessExists(CELL, VALUE);
+
+        when(delegate.get(CELL)).thenReturn(Futures.immediateFuture(Optional.of(VALUE)));
+        assertThat(consensusForgettingStore.get(CELL).get()).contains(VALUE);
+        verify(delegate).get(CELL);
+    }
+
+    @Test
+    public void timesCheckAndTouchOperations() {
+        consensusForgettingStore.checkAndTouch(CELL, VALUE);
+        assertThat(metrics.checkAndTouch().getCount()).isEqualTo(1);
+
+        consensusForgettingStore.checkAndTouch(CELL, VALUE);
+        assertThat(metrics.checkAndTouch().getCount()).isEqualTo(2);
+
+        doAnswer(invocation -> {
+                    Uninterruptibles.sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
+                    return null;
+                })
+                .when(delegate)
+                .checkAndTouch(any(), any());
+        consensusForgettingStore.checkAndTouch(CELL, VALUE);
+        assertThat(metrics.checkAndTouch().getCount()).isEqualTo(3);
+
+        // Metric registries use microseconds internally
+        assertThat(metrics.checkAndTouch().getSnapshot().getMax())
+                .isGreaterThanOrEqualTo(TimeUnit.MILLISECONDS.toMicros(100));
+    }
+
+    @Test
+    public void tracksConcurrentCheckAndTouchOperations() throws BrokenBarrierException, InterruptedException {
+        int iterations = 5;
+        CountDownLatch latch = new CountDownLatch(iterations);
+        CyclicBarrier barrier = new CyclicBarrier(2);
+
+        doAnswer(invocation -> {
+                    latch.countDown();
+                    barrier.await();
+                    latch.await();
+                    return null;
+                })
+                .when(delegate)
+                .checkAndTouch(any(), any());
+
+        ExecutorService executorService = PTExecutors.newFixedThreadPool(iterations);
+        for (int iteration = 1; iteration <= iterations; iteration++) {
+            executorService.execute(() -> consensusForgettingStore.checkAndTouch(CELL, VALUE));
+            barrier.await();
```

Comment:
sorry why do we need the barrier?

Reply:
The barrier is there to stop the thread that's running the main loop from checking that the number of concurrent things increased before the actual checkAndTouch got executed.

Reply:
ah makes sense - I think we should rename the variables still to make it a bit more obvious 😅